stdlib/universe/fill.gen.go

```diff
@@ -0,0 +1,240 @@
+// Generated by tmpl
+// https://github.com/benbjohnson/tmpl
+//
+// DO NOT EDIT!
+// Source: fill.gen.go.tmpl
+
+package universe
+
+import (
+	"fmt"
+
+	"github.com/apache/arrow/go/arrow"
+	"github.com/apache/arrow/go/arrow/array"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/values"
+)
+
+func (t *fillTransformation) fillColumn(typ flux.ColType, arr array.Interface, fillValue *interface{}) array.Interface {
+	switch typ {
+	case flux.TInt:
+		return t.fillIntColumn(arr.(*array.Int64), fillValue)
+	case flux.TUInt:
+		return t.fillUintColumn(arr.(*array.Uint64), fillValue)
+	case flux.TFloat:
+		return t.fillFloatColumn(arr.(*array.Float64), fillValue)
+	case flux.TBool:
+		return t.fillBooleanColumn(arr.(*array.Boolean), fillValue)
+	case flux.TString:
+		return t.fillStringColumn(arr.(*array.Binary), fillValue)
+	case flux.TTime:
+		return t.fillTimeColumn(arr.(*array.Int64), fillValue)
+
+	default:
+		panic(fmt.Errorf("unsupported array data type: %s", arr.DataType()))
+	}
+}
+
+func (t *fillTransformation) fillIntColumn(arr *array.Int64, fillValue *interface{}) array.Interface {
+	fillValueNull := *fillValue == nil
+	var fillValueInt int64
+	if !fillValueNull {
+		fillValueInt = (*fillValue).(int64)
+	} else {
+		if fillValueNull = arr.IsNull(0); !fillValueNull {
```

Comment:
I am not sure there's a circumstance where this is correct. The only thing I can think of is when `UsePrevious` is true and this value gets set lower. I think we can skip this here.

Reply:
This is for handling the case when we have not encountered a non-null value in the table yet and the first value in the current buffer is still `null`. Think of an extreme case when all the rows in the specified table are `null` for the designated column.

Reply:
I don't know what "set lower" really means.

Reply:
Later in the function the fill value gets set when it encounters a non-null value. So this code doesn't seem to add anything and this code would only be valid for the `UsePrevious` case anyway.